src/components/ProgressIndicator/ProgressIndicator.js

```diff
@@ -9,49 +9,113 @@ import PropTypes from 'prop-types';
     [`${prefix}--progress-step--complete`]: complete,
     [`${prefix}--progress-step--incomplete`]: !complete && !current,
+    [`${prefix}--progress-step--disabled`]: disabled,
     [className]: className,
   });
 
-  const currentSvg = current && (
-    <svg>
-      <circle cx="12" cy="12" r="12" />
-      <circle cx="12" cy="12" r="6" />
-      <title>{description}</title>
-    </svg>
-  );
-
-  const completeSvg = complete && (
-    <svg width="16" height="16" viewBox="0 0 16 16">
-      <title>{description}</title>
-      <g fillRule="nonzero">
-        <path d="M8 15A7 7 0 1 0 8 1a7 7 0 0 0 0 14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z" />
-        <path d="M11.646 5.146l.708.708-5.604 5.603-3.104-3.103.708-.708 2.396 2.397z" />
-      </g>
-    </svg>
-  );
-
-  const incompleteSvg = !complete && (
-    <svg>
-      <title>{description}</title>
-      <circle cx="12" cy="12" r="12" />
-    </svg>
-  );
+  const currentSvg =
+    current &&
+    (componentsX ? (
+      <svg>
+        <path d="M 7, 7 m -7, 0 a 7,7 0 1,0 14,0 a 7,7 0 1,0 -14,0" />
+      </svg>
+    ) : (
+      <svg>
+        <circle cx="12" cy="12" r="12" />
+        <circle cx="12" cy="12" r="6" />
+        <title>{description}</title>
+      </svg>
+    ));
+
+  const completeSvg =
+    complete &&
+    (componentsX ? (
+      <CheckmarkOutline16 aria-label={description} role="img" />
+    ) : (
+      <svg width="16" height="16" viewBox="0 0 16 16">
+        <title>{description}</title>
+        <g fillRule="nonzero">
+          <path d="M8 15A7 7 0 1 0 8 1a7 7 0 0 0 0 14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z" />
+          <path d="M11.646 5.146l.708.708-5.604 5.603-3.104-3.103.708-.708 2.396 2.397z" />
+        </g>
+      </svg>
+    ));
+  const incompleteSvg = (() => {
+    if (complete) {
+      return null;
+    }
+    if (componentsX) {
+      if (invalid) {
+        return <Warning16 className={`${prefix}--progress__warning`} />;
+      }
+      return (
+        <svg>
+          <path d="M8 1C4.1 1 1 4.1 1 8s3.1 7 7 7 7-3.1 7-7-3.1-7-7-7zm0 13c-3.3 0-6-2.7-6-6s2.7-6 6-6 6 2.7 6 6-2.7 6-6 6z" />
+        </svg>
+      );
+    }
+    return (
+      <svg>
+        <title>{description}</title>
+        <circle cx="12" cy="12" r="12" />
+      </svg>
+    );
+  })();
 
   return (
     <li className={classes}>
       {currentSvg || completeSvg || incompleteSvg}
-      <p className={`${prefix}--progress-label`}>{label}</p>
+      {componentsX &&
+      overflowTooltipChildren !== undefined &&
+      overflowTooltipChildren !== null ? (
+        <Tooltip
+          direction="bottom"
+          showIcon={false}
+          triggerClassName={`${prefix}--progress-label`}
+          triggerText={label}
+          {...overflowTooltipProps}>
+          {overflowTooltipChildren}
+        </Tooltip>
+      ) : (
+        <p
+          className={`${prefix}--progress-label`}
+          aria-describedby={componentsX ? 'label-tooltip' : null}>
```

Comment:
We make sure we generate instance-speciic ID for ID refs like `aria-describedby`. Which element should this point to BTW?